Segfault when trying to open ingress addon

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

BUG REPORT

**Minikube version** (use `minikube version`):

v0.15

**Environment**:
- Ubuntu 16.04
- VirtualBox
- Docker client v1.12.5, server v1.11

**What happened**:

1. `minikube addons enable ingress`
2. Create a Deployment+Service+Ingress that has a `path` (using kubectl).
2. `minikube addons open ingress`

It blocks for a while with nothing happening. Eventually:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4d7e12]

goroutine 1 [running]:
panic(0x19551e0, 0xc420010100)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
k8s.io/minikube/pkg/util.RetriableError.Error(0x0, 0x0, 0xc4205258e0, 0x42b1ba)
        /var/lib/jenkins/go2/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/pkg/util/utils.go:40 +0x22
k8s.io/minikube/pkg/util.(*RetriableError).Error(0xc420412580, 0x42b2ae, 0x32e4240)
        <autogenerated>:1 +0x5b
k8s.io/minikube/pkg/util.MultiError.ToError(0xc4201eb000, 0x14, 0x20, 0x10, 0x11)
        /var/lib/jenkins/go2/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/pkg/util/utils.go:169 +0x87
k8s.io/minikube/pkg/util.RetryAfter(0x14, 0xc420525c30, 0x165a0bc00, 0x0, 0xc420317d18)
        /var/lib/jenkins/go2/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/pkg/util/utils.go:98 +0x127
k8s.io/minikube/cmd/minikube/cmd/config.glob..func9(0x3d524a0, 0xc420413e60, 0x1, 0x1)
        /var/lib/jenkins/go2/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/cmd/minikube/cmd/config/open.go:93 +0x48f
k8s.io/minikube/vendor/github.com/spf13/cobra.(*Command).execute(0x3d524a0, 0xc420413dd0, 0x1, 0x1, 0x3d524a0, 0xc420413dd0)
        /var/lib/jenkins/go2/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/vendor/github.com/spf13/cobra/command.go:603 +0x439
k8s.io/minikube/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x3d504c0, 0xc4200320e8, 0xc420637ee8, 0x4672aa)
        /var/lib/jenkins/go2/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/vendor/github.com/spf13/cobra/command.go:689 +0x367
k8s.io/minikube/vendor/github.com/spf13/cobra.(*Command).Execute(0x3d504c0, 0x0, 0xc4200320e8)
        /var/lib/jenkins/go2/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/vendor/github.com/spf13/cobra/command.go:648 +0x2b
k8s.io/minikube/cmd/minikube/cmd.Execute()
        /var/lib/jenkins/go2/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/cmd/minikube/cmd/root.go:97 +0x31
main.main()
        /var/lib/jenkins/go2/src/k8s.io/minikube/cmd/minikube/main.go:28 +0x19
```

If you skip step 2 (creating Ingress with path inside of K8s) it just times out instead of segfaulting.

--- a/cmd/minikube/cmd/config/open.go
+++ b/cmd/minikube/cmd/config/open.go
@@ -20,14 +20,12 @@ import (
 	"fmt"
 	"os"
 	"text/template"
-	"time"
 
 	"github.com/docker/machine/libmachine"
 	"github.com/spf13/cobra"
 	"k8s.io/minikube/pkg/minikube/assets"
 	"k8s.io/minikube/pkg/minikube/cluster"
 	"k8s.io/minikube/pkg/minikube/constants"
-	"k8s.io/minikube/pkg/util"
 )
 
 var (
@@ -83,21 +81,17 @@ minikube addons enable %s`, addonName, addonName))
 
 		namespace := "kube-system"
 		key := "kubernetes.io/minikube-addons-endpoint"
-		if err := util.RetryAfter(20,
-			func() error {
-				_, err := cluster.GetServiceListByLabel(namespace, key, addonName)
-				if err != nil {
-					return err
-				}
-				return nil
-			}, 6*time.Second); err != nil {
-			fmt.Fprintf(os.Stderr, "Could not find endpoint for addon %s: %s\n", addonName, err)
-			os.Exit(1)
-		}
+
 		serviceList, err := cluster.GetServiceListByLabel(namespace, key, addonName)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error getting service with namespace :%s and labels %s:%s: %s", namespace, key, addonName, err)
+			fmt.Fprintf(os.Stderr, "Error getting service with namespace: %s and labels %s:%s: %s", namespace, key, addonName, err)
 			os.Exit(1)
+		}
+		if len(serviceList.Items) == 0 {
+			fmt.Fprintf(os.Stdout, `
+This addon does not have an endpoint defined for the 'addons open' command
+You can add one by annotating a service with the label %s:%s`, key, addonName)
+			os.Exit(0)
 		}
 		for i := range serviceList.Items {
 			service := serviceList.Items[i].ObjectMeta.Name

--- a/pkg/minikube/cluster/cluster.go
+++ b/pkg/minikube/cluster/cluster.go
@@ -760,8 +760,6 @@ func getServiceListFromServicesByLabel(services corev1.ServiceInterface, key str
 	if err != nil {
 		return &v1.ServiceList{}, &util.RetriableError{Err: err}
 	}
-	if len(serviceList.Items) == 0 {
-		return &v1.ServiceList{}, &util.RetriableError{Err: err}
-	}
+
 	return serviceList, nil
 }

--- a/pkg/minikube/cluster/cluster_test.go
+++ b/pkg/minikube/cluster/cluster_test.go
@@ -876,8 +876,8 @@ func TestGetServiceListFromServicesByLabel(t *testing.T) {
 	serviceIface := ServiceInterfaceMock{
 		ServiceList: serviceList,
 	}
-	if _, err := getServiceListFromServicesByLabel(serviceIface, "shouldError", "shouldError"); err == nil {
-		t.Fatalf("Service had no label match but getServiceListFromServicesByLabel did not return an error")
+	if _, err := getServiceListFromServicesByLabel(serviceIface, "nothing", "nothing"); err != nil {
+		t.Fatalf("Service had no label match, but getServiceListFromServicesByLabel returned an error")
 	}
 
 	if _, err := getServiceListFromServicesByLabel(serviceIface, "foo", "bar"); err != nil {

